mantidproject/mantid
# Remove the dodgy passing of algorithms in *Dialog methods

re #28093

**Description of work.**
For the *Dialog methods for aglorithms we used to pass the algorithm name that was created in Python, and then on the C++ side we would extract the most recent algorithm of the same name from the AlgorithmManager and hope it was the same one.

This replaces that with just passing the algorithm across the boundary and removes the dodgy hack method.


**To test:**
1. In Mantidplot, make sure the AlgNameDialog, e.g. RebinDialog methods still work and execute successfully.


re #28093. 


*This does not require release notes* because it is purely internal and does not affect the functionality


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
